File: CodingTest/Programmers/BruteForce/studentWhoGiveUpMath.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    
    vector<int> one = {1,2,3,4,5};
    vector<int> two = {2,1,2,3,2,4,2,5};
    vector<int> three = {3,3,1,1,2,2,4,4,5,5};
    
    vector<int> score(3);
    
    // 각각 몇 개씩 맞췄는지 채점
    for (int i = 0; i < answers.size(); i++){
        if (answers[i] == one[i%5])
            score[0]++;
        if (answers[i] == two[i%8])
            score[1]++;
        if (answers[i] == three[i%10])
            score[2]++;
    }
    
    // 최고점수 구하기
    int max_score = *max_element(score.begin(), score.end());
    
    // 최고점수를 얻은 수포자 answer에 넣기
    for (int i = 0; i < 3; i++) {
        if (score[i] == max_score) {
            answer.push_back(i+1);
        }
    }
    
    return answer;
}

File: CodingTest/Programmers/BruteForce/studentWhoGiveUpMath.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int[] answers) {
        int[] answer = {};
        int[] one = {1,2,3,4,5};
        int[] two = {2,1,2,3,2,4,2,5};
        int[] three = {3,3,1,1,2,2,4,4,5,5};
        
        int score1 = 0;
        int score2 = 0;
        int score3 = 0;
        
        for (int i = 0; i < answers.length; i++) {
            if (answers[i] == one[i%5])
                score1++;
            if (answers[i] == two[i%8])
                score2++;
            if (answers[i] == three[i%10])
                score3++;
        }
        
        int max = Math.max(Math.max(score1, score2), score3);        // 최댓값 구하기
        
        ArrayList<Integer> list = new ArrayList<Integer>();
        
        if (max == score1)
            list.add(1);
        if (max == score2)
            list.add(2);
        if (max == score3)
            list.add(3);
        
        answer = new int[list.size()];
        
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }
        
        return answer;
    }
}